MessagesView/MessagesView.swift

```diff
@@ -165,6 +170,24 @@ extension MessagesView : UICollectionViewDelegateFlowLayout {
     }
     
     public func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
-        return CGSize(width: 250, height: 100) //TODO: calculate accurate size
+        guard let message = dataSource?.messages[indexPath.row].text,
+            
+        let cell = MessageCollectionViewCell.fromNib() else {
+            return defaultCellSize
+        }
+        
+        let maxWidth = collectionView.bounds.size.width - collectionView.contentInset.left - collectionView.contentInset.right
+        let cellMargins = cell.layoutMargins.left + cell.layoutMargins.right
+        let requiredWidth = maxWidth - cellMargins
+        
+        let labelMargins = cell.labelLeadingConstraint.constant + cell.backgroundLeadingConstraint.constant + cell.backgroundTrailingConstraint.constant + cell.labelTrailingConstraint.constant
+        
+        cell.textLabel.text = message
+        cell.textLabel.preferredMaxLayoutWidth = requiredWidth - messageMargin - labelMargins
+        cell.labelWidthLayoutConstraint.constant = requiredWidth - messageMargin - labelMargins
+
+        var size = cell.contentView.systemLayoutSizeFitting(UILayoutFittingCompressedSize)
+        size.width = requiredWidth
+        return size
```

Comment:
We should consider moving above calculations into separate function, as it fits only one case: message with text only. This way we are aware that function can be replaced, upgraded or enhanced in future.